File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for idx, num in enumerate(numbers):
            diff = target - numbers[idx]
            right = len(numbers)-1
            left = idx+1
            while left <= right:
                mid = left + (right-left) // 2
                if (numbers[mid] == diff):
                    return [idx+1, mid+1]
                elif numbers[mid] < diff:
                    left = mid+1
                else: 
                    right = mid-1

File: Data Structures & Algorithms/two-integer-sum/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = new Map()
        for (let idx=0; idx<nums.length;idx++){
            let diff = target - nums[idx]
            let checkExist = map.has(diff)
            if (checkExist) return [map.get(diff), idx]
            map.set(nums[idx], idx)
        }
        return [-1, -1]
    }
}
